src/pages/index.js

```diff
@@ -85,43 +127,157 @@ const addModal = new PopUpWithForm(
 editModal.setEventListeners();
 addModal.setEventListeners();
 
+const cardDeletePopUp = new PopUpWithConfirmation({
+  popUpSelector: "#delete-modal",
+});
+
+cardDeletePopUp.setEventListeners();
+
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues);
-  editModal.close();
+  editModal.setLoading(true);
+  api
+    .setUserInfo(inputValues.title, inputValues.description)
+    .then(() => {
+      userInfo.setUserInfo(inputValues);
+      editModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editModal.setLoading(false);
+    });
 }
 
 function handleImageClick(name, link) {
   previewCardModal.open(name, link);
 }
 
+function handleEditButtonClick() {
+  profileEditModal.classList.add("modal_opened");
+}
+
+if (profileEditButton) {
+  profileEditButton.addEventListener("click", handleEditButtonClick);
+}
+
 const previewCardModal = new PopUpWithImage("#modal-preview");
 previewCardModal.setEventListeners();
 
 function createCard(data) {
-  const cardElement = new Card(data, "#card-template", handleImageClick);
+  const cardElement = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
   return cardElement.getView();
 }
 
 function renderCard(cardData) {
-  const addCard = new Card(cardData, cardSelector, handleImageClick);
-  return addCard.getView();
+  const cardInstance = new Card(
+    cardData,
+    cardSelector,
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
+  return cardInstance.getView();
 }
 
 function handleAddCardSubmit(inputValues) {
-  const name = inputValues.name;
-  const link = inputValues.link;
-  const cardEl = renderCard({ name, link });
-  cardSection.addItem(cardEl);
-  addModal.close();
+  addModal.setLoading(true);
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      const cardEl = renderCard(data);
+      section.addItem(cardEl);
+      addModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addModal.setLoading(false);
+    });
 }
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
+function handleAvatarSubmit(url) {
+  profileAvatarPopUp.setLoading(true);
+  api
+    .updateAvatar(url)
+    .then((data) => {
+      userInfo.setAvatar(data);
+      profileAvatarPopUp.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileAvatarPopUp.setLoading(false);
+    });
+}
+
+function handleDeleteCard(cardInstance) {
+  cardDeletePopUp.open();
+  cardDeletePopUp.setSubmitAction(() => {
+    cardDeletePopUp.setLoading(true, "Deleting...");
+    api
+      .deleteCard(cardInstance._data._id)
```

Comment:
`_data`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.